state card for vacuum component with toolbar in more-info-card

azogue opened the pull request:
For the new `vacuum` component ([#8623](https://github.com/home-assistant/home-assistant/pull/8623#issuecomment-317968096)), a first and basic custom card, using the exposed services for these components.

* Basic functionality:

- main switch for turn_on / turn_off.
- ~Show secondary line with optional status, fanspeed and battery status.~
- Show toolbar with play_plause, stop, locate and return_home service calls in the more-info card.
- Use support flags for all extra services (not for turn_on/turn_off).
- State attributes table with the extra attrs added in each vacuum platform.

Screenshots are [here](https://github.com/home-assistant/home-assistant/pull/8623#issuecomment-318138136)

azogue committed:
state card for vacuum component with toolbar in more-info-card

* Basic functionality:

- main switch for turn_on / turn_off.
- Show secondary line with optional status, fanspeed and battery status.
- Show toolbar with play_plause, stop, locate and return_home service calls in the more-info card.
- Use support flags for all extra services (not for turn_on/turn_off).
- Filter specific (vacuum) keys in the state attributes table.

azogue committed:
lint fix

azogue committed:
remove the more-info card, use speed steps with input select

- No need for `more-info-vacuum.html` (using `hidden$='{{inDialog}}'` to separate views).
- Added an input select for the `vacuum_fanspeed_list` property, to set fanspeeds within the more-info view.

This way the controls appear above the history chart, and any extra attribute attached in the platform shows in the normal table below.

azogue committed:
change `start_pause` service

balloob commented on src/state-summary/state-card-vacuum.html:
```diff
@@ -0,0 +1,205 @@
+      <ha-entity-toggle state-obj="[[stateObj]]" hass='[[hass]]'></ha-entity-toggle>
+    </div>
+
+    <div hidden$='{{inDialog}}'>
```
These should all be removed from the state card and be moved to the `more-info` card.

balloob replied:
The state cards are all name + primary control/state. No extra lines are allowed.

azogue replied:
> These should all be removed from the state card and be moved to the more-info card

That was my first try, but I changed it because (for me) it looked better with the controls in top of the history chart, with the attrs table below, but I'll change it to use the more-info card (with the last change I wasn't using it)

> The state cards are all name + primary control/state. No extra lines are allowed.

I took the climate card as example (first time here), and thought that the status, fan speed and battery could be in a line below. So this is not allowed?

<img width="352" alt="captura de pantalla 2017-07-29 a las 9 29 12" src="https://user-images.githubusercontent.com/9748301/28743059-8929b18a-7440-11e7-911b-695b93d9139d.png">

If so, I'll remove it, but I think I'll use my first attempt as custom_UI, I like the idea of seeing more than a switch here.

azogue replied:
@balloob, what do you think about the attrs filter in `window.hassUtil.LOGIC_STATE_ATTRIBUTES`, I wasn't sure of that...

balloob replied:
You can remove them. It's only for domains that do not have a more info card.

balloob replied:
Re: controls above history

It's more important to stay consistent than to have 1 domain look better.

azogue replied:
ok, but then, if I understand, **there is no need for a new state card**, only the **more-info card** is needed, right? Because the default card for the vacuum is a switch, I think.

What about the filtering of state attrs with `window.hassUtil.LOGIC_STATE_ATTRIBUTES`?

balloob replied:
Correct

> What about the filtering of state attrs with `window.hassUtil.LOGIC_STATE_ATTRIBUTES`?

You can remove them. It's only for domains that do not have a more info card.

azogue replied:
ok, so instead of calling `<ha-attributes state-obj="[[stateObj]]"></ha-attributes>` to make the attrs table, I make it manually filtering the keys I want to hide, isn't it?

balloob replied:
Maybe you can extend ha-attributes to take an extra argument with extra filters. That way we don't have to add domain specific knowledge in a generic component.

azogue replied:
it's already done, as I see, in `more-info-light` is calling:
`<ha-attributes state-obj="[[stateObj]]" extra-filters="brightness,color_temp,white_value,effect_list,effect,rgb_color,xy_color,min_mireds,max_mireds"></ha-attributes>`
That's what I need, right? And that way I can rename the attrs to remove the `vacuum_` prefix in the HA PR.

balloob replied:
yes.

azogue replied:
it's done in the last commit, but I'm kind AFK until tuesday (not on my dev machine), so I can't test it.

azogue replied:
@balloob, I changed some things (and tested it), I hope now it's ready.

The final appearance is:
![captura de pantalla 2017-07-30 a las 16 21 07](https://user-images.githubusercontent.com/9748301/28754495-a8483b5c-7546-11e7-9599-9494a549d168.png)

And the state card is the default switch view.

azogue committed:
remove state-card and add more-info card

- No need to make a state-card, as no extra line is allowed in the state-card
- No need to add attrs keys to `LOGIC_STATE_ATTRIBUTES`, using the `extra-filters=` parameter.
- New more-info card with the vacuum controls and the extra state attrs table.

azogue committed:
change state attrs names

azogue committed:
one more change state attrs names

azogue committed:
fix justification

azogue committed:
fix vertical space

azogue committed:
fix hide fan_speed when not supported

balloob reviewed (approved):
Looks good 🐬
